wordpress-mobile/WordPress-iOS
# Domains: full flow.

This should be the last PR of domains (apart from some UI bugs that I'll fix tomorrow, including better iPad layout and what @ScoutHarris noticed in the last PR Review).

This should allow you to redeem a domain and then fall back to a plugin install process seemlessly — allowing Biz users that don't have custom domains to complete the entire flow.

Please not that due to the API (and it's flakiness — I've basically added "oh this failed? that doesn't seem right. lets retry 10 times" logic in multiple places)  this process can take _excruciatingly_ long time — up to 5 minutes wall time. 

To test:

1. Get a Biz domain without domain (ping me if you need credits or to cancel a domain, I have SA superpowers).
2. Try to install a plugin
3. If in doubt, watch console output to see whether everything is going fine
4. Verify that the plugin installed correctly
